src/services/helpers.ts

```diff
@@ -56,3 +58,31 @@ export const shuffleArray = <T>(array: T[]): T[] => {
   }
   return shuffled
 }
+
+export const getNextExercise = async (
+  profession: Discipline
+): Promise<{ disciplineId: number; exerciseKey: number }> => {
+  const disciplines = await loadDisciplines(profession) // TODO LUN-316 leaf disciplines must be loaded, also if nested
+  if (disciplines.length <= 0) {
+    throw new Error(`No Disciplines for id ${profession.id}`)
+  }
+  const progress = await AsyncStorage.getExerciseProgress()
+  const doneExercisesOfLeafDiscipline = (disciplineId: number): number => {
+    const progressOfDiscipline = progress[disciplineId]
+    return progressOfDiscipline
+      ? Object.keys(progressOfDiscipline).filter(item => progressOfDiscipline[item] !== undefined).length
+      : 0
+  }
+  const result = disciplines.find(discipline => doneExercisesOfLeafDiscipline(discipline.id) < exercisesWithProgress)
```

Comment:
Maybe you could rename `result` to what it actually contains (something like `unfinishedDiscipline` probably?

Reply:
Renamed to firstUnfinishedDiscipline